may-2018-reenabled-opcodes.md

```diff
@@ -0,0 +1,428 @@
+The original operations can be implemented with varying combinations of `OP_SPLIT`, `OP_SWAP` and `OP_DROP`.
+
+
+## <a name="data-types"></a>Script data types
+
+It should be noted that in script operation data values on the stack are interpreted as either byte sequences  
+or numeric values.  **All data on the stack is interpreted as a byte sequence unless specifically stated as being interpreted 
+as a numeric value.**
+
+The numeric value type has specific limitations:
+1. The used encoding is little endian with an explicit sign bit (the highest bit of the last byte).
+2. They cannot exceed 4 bytes in length.
+3. They must be encoded using the shortest possible byte length (no zero padding)
+    3. There is one exception to rule 3: if there is more than one byte and the most significant bit of the 
+        second-most-significant-byte is set it would conflict with the sign bit. In this case a single 0x00 or 0x80 byte is allowed
+        to the left.
+4. Zero is encoded as a zero length byte sequence. Single byte positive or negative zero (0x00 or 0x80) are not allowed.
+    
+The new opcode `x OP_BIN2NUM -> out` can be used convert a byte sequence into a numeric value where required.
+
+The new opcode `x n OP_NUM2BIN` can be used to convert a numeric value into a zero padded byte sequence of length `n` 
+whilst preserving the sign bit.
+
+**Endian notation**
+
+For human readability where hex strings are presented in this document big endian notation is used.  That is: 0x0100 represents 
+decimal 256, not decimal 1.
+
+
+## Risks and philosophical approach during design
+
+In general the approach taken is a minimalist one in order limit edge cases as much as possible.  Where it is possible
+for a primitive op code used in conjunction with existing op codes to be combined to produce several more complex 
+operations that is preferred over a set of more complex op codes.  Input conditions that create ambiguous or undefined 
+behaviour should fail fast.  
+
+Each op code should be examined for the following risk conditions and mitigating behaviour defined explicitly:
+* Operand byte length mismatch.  Where it would be normally expected that two operands would be of matching byte lengths
+  the resultant behaviour should be defined.
+* Signed integer.  Whether signed integers are permitted operands and whether any special handling is required.
+* Stack size impact.  Both number of elements and total size of elements. 
+* Overflows.  Defined behaviour in the instance that result of the operation exceeds MAX_SCRIPT_ELEMENT_SIZE
+* Empty byte sequence operands.  Whether empty byte sequences should be allowed as a representation of zero.
+* Empty byte sequence output.  Note that an operation that outputs an empty byte sequence has effectively pushed `false` 
+  onto the stack. If this is the last operation in a script or if a conditional operator immediately follows the script 
```

Comment:
For grammatical clarity, I would rephrase this sentence.

"The script author must consider the possibility that this is the last operation in a script, and the possibility that a conditional operator immediately follows the script."

Reply:
section removed